exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentNodeUtils.java

```diff
@@ -107,49 +105,30 @@ static ServiceInfo getServiceInfo(String serviceName) {
   }
 
   @Nullable
-  static Resource getDefaultResourceProto() {
-    return toResourceProto(RESOURCE);
+  static Resource getAutoDetectedResourceProto() {
+    return toResourceProto(AUTO_DETECTED_RESOURCE);
   }
 
   // Converts a MonitoredResource to a Resource proto.
-  // TODO: update this method to use the Java Resource API once auto-detection is implemented.
   @Nullable
   @VisibleForTesting
-  static Resource toResourceProto(@Nullable MonitoredResource resource) {
-    if (resource == null) {
+  static Resource toResourceProto(@Nullable io.opencensus.resource.Resource resource) {
+    if (resource == null || resource.getType() == null) {
       return null;
     } else {
       Resource.Builder resourceProtoBuilder = Resource.newBuilder();
-      resourceProtoBuilder.setType(resource.getResourceType().name());
+      resourceProtoBuilder.setType(resource.getType());
       putResourceLabels(resource, resourceProtoBuilder);
       return resourceProtoBuilder.build();
     }
   }
 
   // Puts the attributes of MonitoredResource to ResourceProto.
   private static void putResourceLabels(
```

Comment:
I think we can get rid of this method and add logic directly ```toResourceProto```.

Reply:
Removed redundant methods.